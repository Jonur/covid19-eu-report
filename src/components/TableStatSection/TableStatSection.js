import React, { useState } from 'react';
import { bool, string } from 'prop-types';
import classNames from 'classnames';
import { TABLE_VIEW_DATA } from '../../definitions/propTypes';
import { TableView } from '..';
import s from './TableStatSection.module.scss';

const TableStatSection = ({
  ariaLabelledBy,
  countryStatColumnName,
  data,
  dataProp,
  dataPropSecondary,
  sectionTitle,
  sectionSubtitle,
  title,
  positive,
}) => {
  const [expandedSection, setExpandedSection] = useState(false);

  return (
    <section aria-labelledby={ariaLabelledBy} className={s.section}>
      <h2
        id={ariaLabelledBy}
        onClick={() => setExpandedSection(!expandedSection)}
        className={s.sectionTitle}
      >
        {sectionTitle}
        <span className={classNames(s.last24hInfo, { [s.positive]: positive })}>
          {sectionSubtitle}
        </span>
        <span
          className={classNames(s.expandCollapseArrow, {
            [s.expanded]: expandedSection,
          })}
        />
      </h2>
      {expandedSection && (
        <TableView
          countryStatColumnName={countryStatColumnName}
          data={data}
          dataProp={dataProp}
          dataPropSecondary={dataPropSecondary}
          title={title}
          positive={positive}
        />
      )}
    </section>
  );
};

TableStatSection.propTypes = {
  ...TABLE_VIEW_DATA,
  ariaLabelledBy: string.isRequired,
  sectionTitle: string.isRequired,
  sectionSubtitle: string.isRequired,
  positive: bool,
};

TableStatSection.defaultProps = {
  positive: false,
};

export default TableStatSection;
