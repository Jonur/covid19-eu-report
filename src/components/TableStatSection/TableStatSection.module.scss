@import '../../media/colours';

.section {
  margin-bottom: 1.5rem;
  max-width: 850px;
  margin: 0 auto 1.5rem;

  .sectionTitle {
    position: relative;
    font-size: 1.25rem;
    margin: 0;
    border-bottom: 2px solid $border-light;

    .last24hInfo {
      display: block;
      font-size: 0.75rem;
      margin: 0.25rem 0;
      color: $app-danger;
      font-weight: normal;

      &.positive {
        color: $app-success;
      }
    }

    .expandCollapseArrow {
      position: absolute;
      top: 0;
      right: 0.5rem;
      transform: rotate(90deg);
      transition: transform 0.2s linear;

      &.expanded {
        transform: rotate(-90deg);
      }

      &:after {
        content: '\2023';
      }
    }
  }
}
