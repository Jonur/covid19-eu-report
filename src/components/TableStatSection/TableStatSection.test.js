import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import TableStatSection from './TableStatSection';

describe('TableStatSection', () => {
  const renderSetup = (overrideProps = {}) => {
    const props = {
      ariaLabelledBy: 'some-id',
      countryStatColumnName: 'Population',
      data: [
        {
          countryName: 'Italy',
          flagSrc: '32.png',
          totalDeaths: 10023,
          deathsLast24h: 889,
          totalCases: 92472,
          casesLast24h: 5974,
          totalRecovered: 12384,
          recoveredLast24h: 1434,
        },
      ],
      dataProp: 'totalCases',
      dataPropSecondary: 'casesLast24h',
      sectionTitle: 'Total cases',
      sectionSubtitle: '(+ ...)',
      title: 'Some info',
      ...overrideProps,
    };

    return {
      component: render(<TableStatSection {...props} />),
      props,
    };
  };

  test('renders the TableStatSection component correctly', () => {
    const { container } = renderSetup().component;
    expect(container).toMatchSnapshot();
  });

  test('renders the TableStatSection component with positive styled labels', () => {
    const { container } = renderSetup({ positive: true }).component;
    expect(container).toMatchSnapshot();
  });

  test('expands the statistics table when the title is clicked', () => {
    const { component, props } = renderSetup();
    fireEvent.click(component.getByText(props.sectionTitle));
    expect(component.getByText(/Italy/)).toBeInTheDocument();
  });
});
