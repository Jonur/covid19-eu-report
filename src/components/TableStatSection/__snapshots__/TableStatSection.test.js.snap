// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableStatSection renders the TableStatSection component correctly 1`] = `
<div>
  <section
    aria-labelledby="some-id"
    class="section"
  >
    <h2
      class="sectionTitle"
      id="some-id"
    >
      Total cases
      <span
        class="last24hInfo"
      >
        (+ ...)
      </span>
      <span
        class="expandCollapseArrow"
      />
    </h2>
  </section>
</div>
`;

exports[`TableStatSection renders the TableStatSection component with positive styled labels 1`] = `
<div>
  <section
    aria-labelledby="some-id"
    class="section"
  >
    <h2
      class="sectionTitle"
      id="some-id"
    >
      Total cases
      <span
        class="last24hInfo positive"
      >
        (+ ...)
      </span>
      <span
        class="expandCollapseArrow"
      />
    </h2>
  </section>
</div>
`;
